### ISSUE TITLE
Swimmer

### ISSUE THREAD

**Issue by EugeneTeu** [coccoinomane/crabada.py#7]
hey @coccoinomane , thanks so much for this bot. Really good work. Im working on getting it to work with swimmer network, not sure if u have already tested on that? If you havent, im willing to contribute for that

**Comment by coccoinomane** [comment]
Hi @EugeneTeu,

Nice to meet you :-)

I haven't made any testing on Swimmer yet but, by all means, feel free to go ahead!

In the best case scenario, the update will entail just a change of RPC, but we cannot be sure until we try it.

Let me know if you need any help in setting up the bot or otherwise.

Cheers,
Cocco

**Comment by EugeneTeu** [comment]
@coccoinomane took a small crack at this (bot works fine for me for c-chain).

seems `EIP-1559` is disabled on test net for now and they seem to be using a default gas price somewhere 

https://docs.swimmer.network/user-docs/alpha-test-swimmer

have tried sending `{'type': 1, 'chainId': 73771, 'gas': 300000, 'gasPrice': 0, 'nonce': 19}` 

but it seems i get a ` {'code': -32000, 'message': 'must be a contract creator: address 0x....'}` 

likely that im being blocked by the RPC. 

are you aware if for the normal operations will swimmer implement EIP1559?

**Comment by EugeneTeu** [comment]
base fee = `25000000000`

**Comment by nikohd12** [comment]
Swimmer network in a day or two. @EugeneTeu @coccoinomane 🤞

**Comment by vancouverjackluo** [comment]
I am getting myself prepared for the swimmer network as well. I think the API will be the same, but as @EugeneTeu mentioned above, subnet uses a fixed gas fee to start with, so dynamic gas to send tx might not work....time to dig up that old web3.py docs.

**Comment by coccoinomane** [comment]
If the only difference will be fixed-gas fee, then we are set thanks to @EugeneTeu 💪
IMHO, it will still probably take a day or two to flatten the bumps.

**Comment by EugeneTeu** [comment]
@vancouverjackluo @coccoinomane I believe we could just keep the current structure and then just hardcode the gas txn to be 25 (or whatever it is set to) + change contract addresses. 

i tested the bot out on swimmer net and seems to be running fine (mining at least)

**Comment by EugeneTeu** [comment]
@vancouverjackluo @coccoinomane I believe we could just keep the current structure and then just hardcode the gas txn to be 25 (or whatever it is set to) + change contract addresses. 

i tested the bot out on swimmer net and seems to be running fine (mining at least)

**Comment by nikohd12** [comment]
> @vancouverjackluo @coccoinomane I believe we could just keep the current structure and then just hardcode the gas txn to be 25 (or whatever it is set to) + change contract addresses.
> 
> i tested the bot out on swimmer net and seems to be running fine (mining at least)

I guess we could since there is no going back from Swimmer at this point cos they also plan on onboarding more games in the future with TUS being the gas.

**Comment by coccoinomane** [comment]
Glad we waited a bit more before implementing the changes: [Swimmer suspended until May](https://twitter.com/swimmer_network/status/1509276976703365120?s=20&t=vzo1PRYh6OXOrLtgVtUIfA).

It seems Crabada has to [rewrite their entire Swimmer impementation](https://twitter.com/oxtender/status/1509294798578253825).

Let's stay tuned for further news.

**Comment by nikohd12** [comment]
Good call here @coccoinomane and @EugeneTeu.

If I may suggest, we all know AVAX gas prices are so high right now because of looting bots coming back to play and other matter, we may want to implement a **GasFeeCap** function.

I saw this bot coded in **Go** you guys might want to [check this out](https://github.com/coinmasterlisting/crabada-bot)
> Tuning gasPrice, gasTip and GasFeeCap(LegacyTx or eip-1559) to speed up Transactions.

**Comment by coccoinomane** [comment]
Hi @nikohd12 ,

I like the `GasFeeCap` idea!

Would you mind opening a separate issue with a proposal?

If the planets align, I could implement it over the weekend :-)

**Comment by nikohd12** [comment]
> Hi @nikohd12 ,
> 
> I like the `GasFeeCap` idea!
> 
> Would you mind opening a separate issue with a proposal?
> 
> If the planets align, I could implement it over the weekend :-)

Sure bro! Thank you!!!

**Comment by manwe88** [comment]
Hello,
I didn't see any updates for new Swimmer network. Has anyone looked into it?

**Comment by coccoinomane** [comment]
The bot is now compatible with the Swimmer Network!

Just follow the instructions in the tutorial linked [here](https://github.com/coccoinomane/crabada.py/discussions/104) 🙂

**Comment by manwe88** [comment]
Thank you! Something was wrong when I looked ino this repository as it seemed that it was not up to date. By the way, how could you find the new idle game api's url as it was not in crabada's docs?
